core/src/main/java/org/infinispan/cache/impl/TypeConverterDelegatingAdvancedCache.java

```diff
@@ -744,4 +744,12 @@ public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
       return returned;
    }
 
+   @Override
+   public AdvancedCache<K, V> lockAs(Object lockOwner) {
+      AdvancedCache<K, V> returned = super.lockAs(lockOwner);
+      if (returned != this && returned instanceof TypeConverterDelegatingAdvancedCache) {
+         ((TypeConverterDelegatingAdvancedCache) returned).entryFactory = this.entryFactory;
```

Comment:
Check this out (core testsuite seems to pass): https://github.com/rvansa/infinispan/commit/1b71c251ad73870aeed88476dafeb18026fb918a